### ISSUE TITLE
a11y improvements

### ISSUE THREAD

**Issue by rnicholus** [FineUploader/fine-uploader#1329]
There are likely a number of accessibility-related adjustments to be made to Fine Uploader UI, the docs site, and the home page.  In the interest of accommodating any and all users who must interact with Fine Uploader, we should make a concerted effort to address any related deficiencies in this library and associated sites.

Here is a running list of improvements to make (please add comments to cover anything I missed):

- [x] Ensure upload button is fully keyboard accessible in all browsers
- [x] Update default templates to use semantic HTML, aria attributes where semantic HTML isn't possible
- [x] Ensure docs site is accessible
- [x] Ensure demos on home page follow a11y best practices
- [x] Ensure website non-demo content is accessible

**Comment by rnicholus** [comment]
The file input button should be reachable via keyboard (tab) in all browsers, with the possible exception of the current version Safari.  Safari seems to have accessibility issues on all sites, so not much we can do here.  

In IE, two tabs are required to navigate past a file input element.  This is due to the fact that IE treats the file input element as two separate elements.  The first tab hits the file input's text box and you cannot invoke the chooser via keyboard at this point.  The second tab hits the file input's button.  To invoke the chooser, you must hit the space bar.  All other browsers only take one tab to navigate past the file input, and the enter key will invoke the chooser.

Previously, keyboard access to the file input was disabled in IE due to the two-tab issue.  I removed this condition as I believe flawed keyboard access is better than no keyboard access.  

There are other potential solutions to ensure consistent keyboard access across all browsers, but all of them involve using JavaScript to either focus the file input, or click it.  The former trips security checks in Firefox, and the latter prevents the underlying form from being submitted via JavaScript in IE.  So, the current behavior is as good as it's gonna get.

**Comment by rnicholus** [comment]
Our website certainly needs some work (JavaScript embedded in `<script>` tag, global vars everywhere, inline styles everywhere, etc) but the site is much more accessible after my changes.  The templates, demos, and examples have been updated as well.  All that's left is the docs site: up next.  

Changes to the docs site, main site, and code will go out with the next scheduled release, which is 5.2.0 at this time.

**Comment by rnicholus** [comment]
Docs site has been updated.  Once our doc generator is fixed, I'll verify and this will be done and ready for 5.2.

**Comment by rnicholus** [comment]
No a11y errors on the develop docs site according to wave.  I'll need to run the same tests on the website and an uploader instance as well before declaring Fine Uploader WCAG compliant.

**Comment by rnicholus** [comment]
Fine Uploader's a11y work should be complete at this point!  Staged in the develop and site/5.2.0 branches.
